trieste/acquisition/function.py

```diff
@@ -130,6 +133,117 @@ def expected_improvement(model: ModelInterface, eta: tf.Tensor, at: QueryPoints)
+        query_points = self._search_space.sample(self._grid_size)
+        query_points = tf.concat([dataset.query_points,query_points],0)
+        fmean, fvar = model.predict(query_points)
+        fsd = tf.math.sqrt(fvar)
+        idx = tf.math.argmax(-fmean[:tf.shape(dataset.query_points)[0]])
+
+        # Fit Gumbel distriubtion
+        # scaling so that gumbel scale is proportional to IQ range of cdf Pr(y*<z)
+        # find quantiles Pr(y*<y1)=r1 and Pr(y*<y2)=r2
+        right = tf.squeeze(fmean[tf.squeeze(idx)])
+        left = right
+        probf = lambda x: tf.math.exp(tf.math.reduce_sum(tfp.distributions.Normal(tf.cast(0,tf.float64),tf.cast(1,tf.float64)).log_cdf(-(x - fmean) / fsd), axis=0))
+        
+        # get range for binary search
+        i = 0
+        while probf(left) < 0.75:
+            left = 2. ** i * tf.math.reduce_min(fmean - 5. * fsd) + (1. - 2. ** i) * right
+            i += 1
+        i = 0
+        while probf(right) > 0.25:
+            right = -2. ** i * tf.math.reduce_min(fmean - 5. * fsd) + (1. + 2. ** i) * tf.squeeze(fmean[tf.squeeze(idx)])
+            i += 1
+
+        # Binary search for 3 percentiles
+        q1, med, q2 = map(lambda val: bisect(lambda x: probf(x) - val, left, right, maxiter=10000, xtol=0.00001),
+                            [0.25, 0.5, 0.75])
+
+        # solve for gumbel params
+        beta = (q1 - q2) / (tf.math.log(tf.math.log(4. / 3.)) - tf.math.log(tf.math.log(4.)))
+        alpha = med + beta * tf.math.log(tf.math.log(2.))
+
+        # sample K length vector from unif([0,1])
+        # return K Y* samples
+        samples = -tf.math.log(-tf.math.log(tf.random.uniform([self._num_samples],dtype=tf.dtypes.float64))) * tf.cast(beta,tf.float64) + tf.cast(alpha,tf.float64)
+
+
+        return lambda at: self._acquisition_function(model, samples, at)
+
+    @staticmethod
+    def _acquisition_function(model: ModelInterface, samples: tf.Tensor, at: QueryPoints) -> tf.Tensor:
+        return max_value_entropy_search(model, samples, at)
+
+
+def max_value_entropy_search(model: ModelInterface, samples: tf.Tensor, at: QueryPoints) -> tf.Tensor:
+    r"""
+    Computes the information gain, i.e the change in entropy of p_min if we would evaluate x.
+
+    See the following for details:
+
+    ::
+
+        @article{wang2017max,
+          title={Max-value entropy search for efficient Bayesian optimization},
+          author={Wang, Zi and Jegelka, Stefanie},
+          journal={arXiv preprint arXiv:1703.01968},
+          year={2017}
+        }
+
+    :param model: The model of the objective function.
+    :param samples: Samples from p_min
+    :param at: The points for which to calculate the expected improvement.
+    :return: The expected improvement at ``at``.
+    """
+    fmean, fvar = model.predict(at)
+    fsd = tf.math.sqrt(fvar)
+    
+    # clip below to improve numerical stability
+    fsd = tf.clip_by_value(fsd, 1.0e-8, tf.float64.max)
```

Comment:
on the other hand, i think these comments are useful. Can you put them on the same line so as to prevent drift? Like
```suggestion
    fsd = tf.clip_by_value(fsd, 1.0e-8, tf.float64.max)  # clip below to improve numerical stability
```

Reply:
(that is, if they fit)